### PR TITLE
MAde it so it saves, (StickWolf need to do her part for this to fully finish)

### PR DESCRIPTION
**Pull request by XanderSuFromQBIX**
Added save-ing

### CODE REVIEW

**Review by XanderSuFromQBIX — COMMENTED** [review]
`Yes, I know, there are accidental line returns, let me fix that. `

**Review by Materia-xx — COMMENTED** [review]
If CurrentGameState is not static that means we can have multiple current game states ?

File: GameEngine/Engine.cs [hunk 1]
```diff
@@ -8,7 +8,8 @@ namespace GameEngine
 {
     public class Engine
     {
-        string NewPlayerName;
+        private GameState CurrentGameState { get; set; }
```
Comment (Materia-xx) [inline GameEngine/Engine.cs]: If CurrentGameState is not static that means we can have multiple current game states ?

File: GameEngine/Engine.cs [hunk 2]
```diff
@@ -44,7 +45,39 @@ public static void Start(Func<IGameData> gameDataFactory, int loadFromSlot = 0)
                 {
                     // Otherwise if we're not starting a new game, show the game intro text
                     // Show the game introduction
-                    engineInstance.StartNewGame();
+                    int displaycount = 1;
+                    Dictionary<int, string> SlotNames = new Dictionary<int, string>();
+                    Console.WriteLine("Hello, wellcome to our game, to load or start a new game, pick a number.");
```
Comment (Materia-xx) [inline GameEngine/Engine.cs]: nit: welcome
Actually IMO I think this is too personalized for the engine to be saying. If there was anything that said "our game" it should be coming from the GameData. Let's make the engine more generic than this.

**Review by Materia-xx — COMMENTED** [review]
nit: welcome

**Review by Materia-xx — COMMENTED** [review]
nit: extra spaces

File: GameEngine/Engine.cs [hunk 3]
```diff
@@ -44,7 +45,39 @@ public static void Start(Func<IGameData> gameDataFactory, int loadFromSlot = 0)
...
+                    Console.WriteLine();
+                    Console.WriteLine($"{displaycount}. Start new Game");
+                    Console.WriteLine("-----------------------------------------");
+                    Console.Write("> ");
+                    int input = int.Parse(Console.ReadLine());
+                    if (SlotNames.ContainsKey(input))
+                    {
+                        string slotToFind;
+                        slotToFind = SlotNames[input];
+                        Console.WriteLine($"Loading {slotToFind}.......");
```
Comment (Materia-xx) [inline GameEngine/Engine.cs]: nit: extra spaces

**Review by Materia-xx — COMMENTED** [review]
But, all it did here was 1 readline, it looks like the call to actually load is below here, so how could it be done loading up here?

File: GameEngine/Engine.cs [hunk 4]
```diff
@@ -44,7 +45,39 @@ public static void Start(Func<IGameData> gameDataFactory, int loadFromSlot = 0)
...
+                    Console.WriteLine("-----------------------------------------");
+                    Console.Write("> ");
+                    int input = int.Parse(Console.ReadLine());
+                    if (SlotNames.ContainsKey(input))
+                    {
+                        string slotToFind;
+                        slotToFind = SlotNames[input];
+                        Console.WriteLine($"Loading {slotToFind}.......");                 
+                        Console.ReadLine();
+                        Console.WriteLine($"Completed Loading {slotToFind}.");
```
Comment (Materia-xx) [inline GameEngine/Engine.cs:70]: But, all it did here was 1 readline, it looks like the call to actually load is below here, so how could it be done loading up here?

**Review by Materia-xx — COMMENTED** [review]
What are all of these extra readlines for ?

File: GameEngine/Engine.cs [hunk 5]
```diff
@@ -44,7 +45,39 @@ public static void Start(Func<IGameData> gameDataFactory, int loadFromSlot = 0)
...
+                        Console.ReadLine();
```
Comment (Materia-xx) [inline GameEngine/Engine.cs]: What are all of these extra readlines for ?

**Review by Materia-xx — COMMENTED** [review]
Looks like you may be changing this file even through you didn't want to. Make sure to look at the files in your PR too.

File: GameEngine/GameState.cs [hunk 6]
```diff
@@ -24,7 +24,6 @@ public static List<string> GetValidSaveSlotNames()
         public static GameState LoadGameState(string slotName)
         {
             // This function should load the named slotName into a GameState and return it.
-
             throw new NotImplementedException();
```
Comment (Materia-xx) [inline GameEngine/GameState.cs:26]: Looks like you may be changing this file even through you didn't want to. Make sure to look at the files in your PR too.

**Review by Materia-xx — COMMENTED** [review]
Doesn't the LoadGameState return the loaded game state ? It doesn't look like you are doing anything with that here.

File: GameEngine/Engine.cs [hunk 7]
```diff
@@ -44,7 +45,39 @@ public static void Start(Func<IGameData> gameDataFactory, int loadFromSlot = 0)
...
+                        GameState.LoadGameState(slotToFind);
```
Comment (Materia-xx) [inline GameEngine/Engine.cs]: Doesn't the LoadGameState return the loaded game state ? It doesn't look like you are doing anything with that here.

**Review by Materia-xx — CHANGES_REQUESTED** [review]
Waiting for updates

File: GameEngine/GameState.cs (hunk 6 above)
Comment (XanderSuFromQBIX) [reply GameEngine/GameState.cs:26]: That was for testing, I changed it back after, NOTHING IS DIFFERENT IN THAT FILE.

Comment (XanderSuFromQBIX) [reply GameEngine/GameState.cs:26]: and I can't do what you are asking for now, it is **"unreverseable"** . I cant change it back now.

File: GameEngine/Engine.cs (hunk 7 above)
Comment (XanderSuFromQBIX) [reply GameEngine/Engine.cs]: i need help with what you are asking here.

**Review by Materia-xx — APPROVED** [review]

**Review by StickWolf — APPROVED** [review]

**Pull request merged**